# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBIndex.cpp
# Commit: 5be5a3bf0499
# Full Hash: 5be5a3bf0499f5501bf1ed6291ba1ac677eacbc6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 16:05:23
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Reduce code duplication for different open cursor variants. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40956
# ==============================================================================

diff -r 6078d5a9c010 -r 5be5a3bf0499 dom/indexedDB/IDBIndex.cpp
--- a/dom/indexedDB/IDBIndex.cpp	Tue Sep 10 15:36:09 2019 +0000
+++ b/dom/indexedDB/IDBIndex.cpp	Tue Sep 10 15:36:15 2019 +0000
@@ -455,28 +455,12 @@
 
   IDBCursor::Direction direction = IDBCursor::ConvertDirection(aDirection);
 
-  OpenCursorParams params;
-  if (aKeysOnly) {
-    IndexOpenKeyCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
+  const CommonIndexOpenCursorParams commonIndexParams = {
+      {objectStoreId, std::move(optionalKeyRange), direction}, indexId};
 
-    params = std::move(openParams);
-  } else {
-    IndexOpenCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  }
+  const auto params =
+      aKeysOnly ? OpenCursorParams{IndexOpenKeyCursorParams{commonIndexParams}}
+                : OpenCursorParams{IndexOpenCursorParams{commonIndexParams}};
 
   RefPtr<IDBRequest> request = GenerateRequest(aCx, this);
   MOZ_ASSERT(request);